src/Illuminate/Encryption/KeyStoreFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+
+namespace Illuminate\Encryption;
+
+
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Contracts\Encryption\KeyStoreFactory as KeyStoreFactoryContract;
+use Illuminate\Encryption\KeyStoreFactory\AppKeyKeyStoreFactory;
+
+class KeyStoreFactory implements KeyStoreFactoryContract
+{
+    private const PATH_TO_KEY_PROVIDER = 'app.encryption.keyProvider';
+
+    public function make(Repository $config): KeyStore
+    {
+        if ($config->get(self::PATH_TO_KEY_PROVIDER)) {
+            return $this->makeKeyStoreFromKeyProvider($config->get(self::PATH_TO_KEY_PROVIDER))
```

Comment:
Consider making `$config->get(self::PATH_TO_KEY_PROVIDER)` into a variable for clarity.

Reply:
Also, consider checking that the result of `app()` is an instance of `KeyStoreFactory`. Otherwise it could lead to really weird errors if somebody forgets to implement that interface.